### PR TITLE
Mrc 3075 - Download comparison report

### PR DESCRIPTION
**Pull request by LekanAnni**
This PR adds download button for comparison output. Please note that this feature is behind feature switch.

**Please kindly test if you can download all output files.**

NB: PR based off mrc-3074, depends on it to be merged before review. 

## Type of version change
Minor 

## Checklist

- [x] I have incremented version number, or version needs no increment
- [x] The build passed successfully, or failed because of ADR tests

### CODE REVIEW

File: src/app/static/src/app/store/downloadResults/actions.ts [hunk 1]
```diff
@@ -10,18 +10,20 @@ export interface DownloadResultsActions {
...
-    prepareOutputs: (store: ActionContext<DownloadResultsState, RootState>) => void
+    prepareComparisonOutput: (store: ActionContext<DownloadResultsState, RootState>, featureSwitch: boolean) => void
+    prepareOutputs: (store: ActionContext<DownloadResultsState, RootState>, featureSwitch: boolean) => void
     poll: (store: ActionContext<DownloadResultsState, RootState>, downloadType: string) => void
 }
 
 export const actions: ActionTree<DownloadResultsState, RootState> & DownloadResultsActions = {
 
-    async prepareOutputs(context) {
+    async prepareOutputs(context, featureSwitch ) {
```
[the remark this answers is not in the document]
Comment (LekanAnni) [reply src/app/static/src/app/store/downloadResults/actions.ts:20]: I Agree but I guess it does complicate testing bit, I went for minimal testing setup approach. It's just a case of removing featureSwitch when we no longer need it.

[the remark this answers is not in the document]
Comment (EmmaLRussell) [reply src/app/static/src/app/store/downloadResults/actions.ts:20]: Fair enough if you don't want `actions` to read the switch. But i still think it's not necessary for both `prepareOutputs` and `prepareComparisonOutput` to have the parameter. Just don't include that last dispatch if featureSwitch is false. That should simplify the testing as won't need to test the two cases for `prepareComparisonOutput`.

**Review by EmmaLRussell — COMMENTED** [review]
Works fine, all files download ok. I think you've accidentally committed a change to enable the feature by default though?

image


Wasn't quite the expecting this change in the actions, I thought you were going to keep the switch param in the top level action and handle not calling the downstream one there. But actually, I think it's fine to just always generate all files from now on, it won't change the UX. This way there's fewer changes required when the switch is removed.

**Review by EmmaLRussell — APPROVED** [review]

File: src/app/static/src/app/components/downloadResults/DownloadResults.vue [hunk 2]
```diff
@@ -92,7 +99,8 @@
     interface ComputedFromDownloadResults {
         spectrum: DownloadResultsDependency,
         coarseOutput: DownloadResultsDependency,
-        summary: DownloadResultsDependency
+        summary: DownloadResultsDependency,
+        comparison: DownloadResultsDependency
```
Comment (Nicolas-Dolan) [inline src/app/static/src/app/components/downloadResults/DownloadResults.vue:103]: So I'm guessing at this point we are intentionally downloading the summary report in place of the comparison report?

**Review by Nicolas-Dolan — APPROVED** [review]
All files are downloading but wasn't sure if downloading the summary report with the comparison report button was intentional

File: src/app/static/src/app/components/downloadResults/DownloadResults.vue (hunk 2 above)
Comment (LekanAnni) [reply src/app/static/src/app/components/downloadResults/DownloadResults.vue:103]: Yes it’s intentional

**Pull request merged**